### PR TITLE
Move risc0-circuit-bigint to risc0 repo

### PR DESCRIPTION
**Pull request by shkoo**

### CODE REVIEW

File: risc0/circuit/bigint/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+[dev-dependencies]
+env_logger = "0.10"
+num-bigint = { version = "0.4", default-features = false, features = ["serde"] }
+
+[target.'cfg(not(target_os = "zkvm"))'.dev-dependencies]
+risc0-circuit-bigint-test-methods = { path = "methods" }
+
+[build-dependencies]
+cc = { version = "1.0", features = ["parallel"] }
+glob = "0.3"
```
Comment (flaub) [inline risc0/circuit/bigint/Cargo.toml:39]: Why are these needed?

File: risc0/circuit/bigint/src/bin/make_control_ids.rs
Comment (flaub) [inline risc0/circuit/bigint/src/bin/make_control_ids.rs:1]: This should be moved to the xtask program.

File: risc0/circuit/bigint/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    pub unconstrained_eval_fn: fn(&mut BigIntContext) -> anyhow::Result<()>,
+    // Control ID of this program
+    pub control_id: Digest,
+    // Control root of a merkle tree with a single element containing control_id
+    pub control_root: Digest,
+    // Number of claims to be verified per invocation of the ZKR
+    pub iters: usize,
+}
+
+pub mod rsa;
```
Comment (flaub) [inline risc0/circuit/bigint/src/lib.rs:80]: This should be moved up to the top with the other mods

File: risc0/circuit/bigint/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    );
+    Ok(claim_refs)
+}
+
+pub(crate) mod generated {
+    #![allow(dead_code)]
+    #![allow(clippy::all)]
+
+    use crate::codegen_prelude::*;
+    include! {"bigint.rs.inc"}
```
Comment (flaub) [inline risc0/circuit/bigint/src/lib.rs:160]: It would help vscode if we could ensure the suffix ends in `.rs`

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -427,15 +427,16 @@ jobs:
...
           version: "0.31.0"
       - name: run cargo semverchecks on ${{ github.event.pull_request.base.sha }}
-        # There seems to be a bug in cargo-semver-checks that can't locate the platform crate...
-        # All other excluded crates aren't published and therefore do not need this check
         run: |
+          # TODO: remove risc0-circuit-bigint exclusion once it exists in main and we
+          # can successfully compare semvers.
           cargo semver-checks --default-features --baseline-rev ${{ github.event.pull_request.base.sha }} \
-            --exclude risc0-zkvm-platform \
+            --exclude risc0-circuit-bigint \
```
Comment (flaub) [inline .github/workflows/main.yml:434]: It's ok if this check fails, it's not a requirement for landing PRs.

File: risc0/circuit/bigint/src/bin/make_control_ids.rs
Comment (flaub) [reply risc0/circuit/bigint/src/bin/make_control_ids.rs:1]: We can do this after this lands, no need for it to be blocked

File: risc0/circuit/bigint/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+[package]
+name = "risc0-circuit-bigint"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+anyhow = "1.0"
+bytemuck = "1.16.1"
+clap = { version = "4.1", features = ["derive"] }
```
Comment (flaub) [inline risc0/circuit/bigint/Cargo.toml:9]: I think this is a `dev-dependency`?

File: risc0/circuit/bigint/src/zkr.rs [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    Ok(Program {
+        code: u32s.iter().cloned().map(BabyBearElem::from).collect(),
+        code_size,
+        po2,
+    })
+}
+
+static REGISTER_ZKRS: std::sync::Once = std::sync::Once::new();
+
+pub fn register_zkrs() {
```
Comment (flaub) [inline risc0/circuit/bigint/src/zkr.rs:62]: When/how does this get called?

File: risc0/circuit/bigint/Cargo.toml (hunk 1 above)
Comment (shkoo) [reply risc0/circuit/bigint/Cargo.toml:39]: removed

File: risc0/circuit/bigint/src/zkr.rs (hunk 6 above)
Comment (shkoo) [reply risc0/circuit/bigint/src/zkr.rs:62]: https://github.com/risc0/risc0/blob/50e505c828d41391fe654c31da753452d73aeec2/risc0/circuit/bigint/src/rsa_tests.rs#L300

File: risc0/circuit/bigint/Cargo.toml (hunk 5 above)
Comment (shkoo) [reply risc0/circuit/bigint/Cargo.toml]: Nope, but we can put it behind a feature

**Review by flaub — APPROVED** [review]

**Review by tzerrell — APPROVED** [review]

File: risc0/circuit/bigint/Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+num-traits = "0.2"
+paste = "1.0"
+pretty_assertions = "1.4"
+risc0-binfmt = { workspace = true }
+risc0-circuit-recursion = { workspace = true, default-features = false }
+risc0-core = { workspace = true }
+risc0-zkp = { workspace = true, features = ["default"] }
+risc0-zkvm = { workspace = true, features = ["std", "unstable"], default-features = false }
+risc0-zkvm-platform = { workspace = true }
+test-log = { version = "0.2", features = ["trace"] }
```
Comment (flaub) [inline risc0/circuit/bigint/Cargo.toml:23]: This is probably a `dev-dependency`

File: risc0/circuit/bigint/Cargo.toml [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+tracing = "0.1.40"
+tracing-subscriber = { version = "0.3.18", features = ["env-filter"] }
```
Comment (flaub) [inline risc0/circuit/bigint/Cargo.toml:25]: this is probably optional on `make_control_ids`
